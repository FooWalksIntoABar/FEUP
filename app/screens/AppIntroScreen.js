import React, { Component } from 'react'
import AppIntroSlider from 'react-native-app-intro-slider'
import PropTypes from 'prop-types'
import Images from '../assets/images/images'

const slides = [
  {
    key: '1',
    title: 'Texto explicativo #1',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#59b2ab'
  },
  {
    key: '2',
    title: 'Texto explicativo #2',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#febe29'
  },
  {
    key: '3',
    title: 'Texto explicativo #3',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#22bcb5'
  }
]

export default class AppIntroScreen extends Component {
  _onDone = () => {
    this.props.navigation.goBack()
  }

  render () {
    return (
      <AppIntroSlider slides={slides} onDone={this._onDone} />
    )
  }
}

AppIntroScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
