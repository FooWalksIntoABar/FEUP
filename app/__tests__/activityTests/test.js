import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import { ActivityScreen } from '../../screens/ActivityScreen'

describe('ActivityScreen snapshot', () => {
  const activities = [
    {
      title: 'Fazer a cama',
      image: 'bed',
      photo: 'bedroom',
      color: '#7d84b2',
      time: {
        min: 0,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Lavar os dentes',
      image: 'toothbrush',
      color: '#0e79b2',
      time: {
        min: 1,
        max: 15,
        goal: 10
      }
    },
    {
      title: 'Vestir',
      image: 'socks',
      color: '#7fb800',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Tomar banho',
      image: 'shower',
      color: '#37c1f0',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Preparar a mochila',
      image: 'bag',
      color: '#e43f6f',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'CalÃ§ar os sapatos',
      image: 'sneakers',
      color: '#4bb3fd',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Tomar o pequeno almoÃ§o',
      image: 'breakfast',
      color: '#ff7f11',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Pentear cabelo',
      image: 'comb',
      color: '#b0db43',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    }
  ]

  jest.useFakeTimers()

  it('renders ActivityScreen correctly with bar & without timer & with vibration feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'bar'}
      activityFeedback={'vibration'}
      showTimer={false}
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap1.js.snap')
  })

  it('renders ActivityScreen correctly with bar & without timer & with visual feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'bar'}
      activityFeedback={'visual'}
      showTimer={false}
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap2.js.snap')
  })

  it('renders ActivityScreen correctly with bar & without timer & with sound feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'bar'}
      activityFeedback={'sound'}
      showTimer={false}
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap3.js.snap')
  })

  it('renders ActivityScreen correctly with bar & with timer & with vibration feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'bar'}
      activityFeedback={'vibration'}
      showTimer
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap4.js.snap')
  })

  it('renders ActivityScreen correctly with bar & with timer & with visual feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'bar'}
      activityFeedback={'visual'}
      showTimer
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap5.js.snap')
  })

  it('renders ActivityScreen correctly with bar & with timer & with sound feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'bar'}
      activityFeedback={'sound'}
      showTimer
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap6.js.snap')
  })

  it('renders ActivityScreen correctly with clock & without timer & with vibration feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'clock'}
      activityFeedback={'vibration'}
      showTimer={false}
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap7.js.snap')
  })

  it('renders ActivityScreen correctly with clock & without timer & with visual feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'clock'}
      activityFeedback={'visual'}
      showTimer={false}
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap8.js.snap')
  })

  it('renders ActivityScreen correctly with clock & without timer & with sound feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'clock'}
      activityFeedback={'sound'}
      showTimer={false}
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap9.js.snap')
  })

  it('renders ActivityScreen correctly with clock & with timer & with vibration feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'clock'}
      activityFeedback={'vibration'}
      showTimer
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap10.js.snap')
  })

  it('renders ActivityScreen correctly with clock & with timer & with visual feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'clock'}
      activityFeedback={'visual'}
      showTimer
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap11.js.snap')
  })

  it('renders ActivityScreen correctly with clock & with timer & with sound feedback', async () => {
    const tree = renderer.create(<ActivityScreen
      navigation={{}}
      progressType={'clock'}
      activityFeedback={'sound'}
      showTimer
      currentActivity={0}
      activities={activities}
      activity={activities[0]}
      setActivityStatus={jest.fn()}
      nextActivity={jest.fn()}
      addStars={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot('./__snapshots__/snap12.js.snap')
  })
})
