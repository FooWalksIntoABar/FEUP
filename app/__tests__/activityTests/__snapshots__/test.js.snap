// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityScreen snapshot renders ActivityScreen correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.6)",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 15,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 0,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 40.019999999999996,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <ProgressBar
      activityFeedback="vibration"
      activityTimes={
        Object {
          "goal": 60,
          "max": 120,
          "min": 0,
        }
      }
      elapsedTime={0}
      feedbackFrequency="normal"
      isPaused={false}
      playSounds={false}
      showTimer={false}
    />
    <CancelButton
      cancelActivity={[Function]}
    />
    <PauseButton
      isPaused={false}
      pauseActivity={[Function]}
      resumeActivity={[Function]}
    />
    <CompleteButton
      completeActivity={[Function]}
      isCompletable={false}
    />
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;

exports[`ActivityScreen snapshot renders ActivityScreen correctly 2`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.6)",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 15,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 0,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 40.019999999999996,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <ProgressBar
      activityFeedback="vibration"
      activityTimes={
        Object {
          "goal": 60,
          "max": 120,
          "min": 0,
        }
      }
      elapsedTime={0}
      feedbackFrequency="normal"
      isPaused={false}
      playSounds={false}
      showTimer={true}
    />
    <CancelButton
      cancelActivity={[Function]}
    />
    <PauseButton
      isPaused={false}
      pauseActivity={[Function]}
      resumeActivity={[Function]}
    />
    <CompleteButton
      completeActivity={[Function]}
      isCompletable={false}
    />
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;

exports[`ActivityScreen snapshot renders ActivityScreen correctly 3`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.6)",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 15,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <ProgressClock
    activityFeedback="vibration"
    activityTimes={
      Object {
        "goal": 60,
        "max": 120,
        "min": 0,
      }
    }
    elapsedTime={0}
    feedbackFrequency="normal"
    isPaused={false}
    playSounds={false}
    showTimer={true}
  />
  <Component
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 0,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 40.019999999999996,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <CancelButton
      cancelActivity={[Function]}
    />
    <PauseButton
      isPaused={false}
      pauseActivity={[Function]}
      resumeActivity={[Function]}
    />
    <CompleteButton
      completeActivity={[Function]}
      isCompletable={false}
    />
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;

exports[`ActivityScreen snapshot renders ActivityScreen correctly 4`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.6)",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 15,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <ProgressClock
    activityFeedback="vibration"
    activityTimes={
      Object {
        "goal": 60,
        "max": 120,
        "min": 0,
      }
    }
    elapsedTime={0}
    feedbackFrequency="normal"
    isPaused={false}
    playSounds={false}
    showTimer={true}
  />
  <Component
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 0,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 40.019999999999996,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <CancelButton
      cancelActivity={[Function]}
    />
    <PauseButton
      isPaused={false}
      pauseActivity={[Function]}
      resumeActivity={[Function]}
    />
    <CompleteButton
      completeActivity={[Function]}
      isCompletable={false}
    />
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;

exports[`ActivityScreen snapshot renders ActivityScreen correctly 5`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="center"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "60%",
        "position": "absolute",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.2)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 10,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;

exports[`ActivityScreen snapshot renders ActivityScreen correctly 6`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="center"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "60%",
        "position": "absolute",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.2)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 10,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;

exports[`ActivityScreen snapshot renders ActivityScreen correctly 7`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#7d84b2",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Image
    resizeMode="center"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": "60%",
        "position": "absolute",
      }
    }
  />
  <Component
    style={
      Object {
        "padding": 40.019999999999996,
        "paddingLeft": 72.036,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 50,
          "textShadowColor": "rgba(0, 0, 0, 0.2)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 10,
        }
      }
    >
      Fazer a cama
    </Text>
  </Component>
  <RewardsModal
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    backPress={[Function]}
    currentActivity={0}
    level={4}
    nextPress={[Function]}
    playSounds={false}
    xp={430}
  />
</Component>
`;
